dali/operators/math/expressions/expression_impl_gpu_binary.cuh

```diff
@@ -20,134 +20,75 @@
-__device__ void ExecuteBinOpND(Result *result, const Left *l, const Right *r, int64_t offset,
-                               int64_t extent, const fast_div<uint64_t> *strides_out,
-                               const int64_t *strides_l, const int64_t *strides_r) {
-  using meta_t = arithm_meta<op, GPUBackend>;
-  int64_t block_start = offset + static_cast<int64_t>(blockDim.x) * blockIdx.x + threadIdx.x;
-  int64_t block_end = offset + extent;
+  const auto &tile = tiles[blockIdx.y];
+  const auto &sample = samples[tile.sample_idx];
+  auto output = static_cast<Result *>(sample.output.data);
+  auto &out_strides = sample.output.strides;
+  auto left = static_cast<const Left *>(sample.args[0].data);
+  auto &left_strides = sample.args[0].strides;
+  auto right = static_cast<const Right *>(sample.args[1].data);
+  auto &right_strides = sample.args[1].strides;
+
+  int ndim = sample.ndim;
+  int64_t block_start = tile.offset + static_cast<int64_t>(blockDim.x) * blockIdx.x + threadIdx.x;
+  int64_t block_end = tile.offset + tile.size;
   int64_t block_step  = static_cast<int64_t>(blockDim.x) * gridDim.x;
 
   for (int64_t idx = block_start; idx < block_end; idx += block_step) {
     uint64_t idx_l = 0, idx_r = 0;
     uint64_t tmp_idx = idx;
-    #pragma unroll
-    for (int d = 0; d < Dims; d++) {
-      int i_d = div_mod(tmp_idx, tmp_idx, strides_out[d]);
-      idx_l += i_d * strides_l[d];
-      idx_r += i_d * strides_r[d];
-    }
-    result[idx] = meta_t::impl(l[idx_l], r[idx_r]);
-  }
-}
 
-/**
- * @brief Loop over tile of `extent` length, binary op with scalar on the left
- */
-template <ArithmeticOp op, typename Result, typename Left, typename Right>
-__device__ void ExecuteBinOp(Result *result, Left l, const Right *r, int64_t offset,
-                             int64_t extent) {
-  using meta_t = arithm_meta<op, GPUBackend>;
-  int64_t start = offset + static_cast<int64_t>(blockDim.x) * blockIdx.x + threadIdx.x;
-  int64_t stride = static_cast<int64_t>(blockDim.x) * gridDim.x;
-  auto *tile_end = result + offset + extent;
-  result += start;
-  r += start;
-  while (result < tile_end) {
-    *result = meta_t::impl(l, *r);
-    result += stride;
-    r += stride;
-  }
-}
-
-/**
- * @brief Loop over tile of `extent` length, binary op with scalar on the right
- */
-template <ArithmeticOp op, typename Result, typename Left, typename Right>
-__device__ void ExecuteBinOp(Result *result, const Left *l, Right r, int64_t offset,
-                             int64_t extent) {
-  using meta_t = arithm_meta<op, GPUBackend>;
-  int64_t start = offset + static_cast<int64_t>(blockDim.x) * blockIdx.x + threadIdx.x;
-  int64_t stride = static_cast<int64_t>(blockDim.x) * gridDim.x;
-  auto *tile_end = result + offset + extent;
-  result += start;
-  l += start;
-  while (result < tile_end) {
-    *result = meta_t::impl(*l, r);
-    result += stride;
-    l += stride;
+    #pragma unroll 6
```

Comment:
Rather a question than a remark: How does work dynamic ndim? Is it equvalent to switch over 7 (0..6) possible bodies or more like sequence of 6 ifs? Or something else?